src/daemon/distribution.c

```diff
@@ -0,0 +1,148 @@
+#include <math.h>
+#include "collectd.h"
+#include "distribution.h"
+
+struct bucket_s{
+	size_t bucket_counter;
+	double min_boundary;
+	double max_boundary;
+};
+
+struct distribution_s {
+     bucket_t *buckets;
+     size_t num_buckets;
+     size_t total_scalar_count;
```

Comment:
This counter should be an `uint64_t`, too.

Reply:
Okay.